genbindings.sh

```diff
@@ -43,45 +38,50 @@ OUT_F="$(pwd)/lightning-c-bindings/include/rust_types.h"
 OUT_CPP="$(pwd)/lightning-c-bindings/include/lightningpp.hpp"
 BIN="$(pwd)/c-bindings-gen/target/release/c-bindings-gen"
 
-pushd "$LIGHTNING_PATH/lightning"
-RUSTC_BOOTSTRAP=1 cargo rustc $FEATURES_ARGS --profile=check -- -Zunstable-options --pretty=expanded > /tmp/lightning-crate-source.txt
-popd
-
 HOST_PLATFORM="$(rustc --version --verbose | grep "host:")"
-if [ "$HOST_PLATFORM" = "host: x86_64-apple-darwin" ]; then
-	# OSX sed is for some reason not compatible with GNU sed
-	sed -E -i '' 's/#!\[crate_name = "(.*)"\]/pub mod \1 {/' /tmp/lightning-crate-source.txt
-else
-	sed -E -i 's/#!\[crate_name = "(.*)"\]/pub mod \1 {/' /tmp/lightning-crate-source.txt
-fi
-echo "}" >> /tmp/lightning-crate-source.txt
 
-if [ "$2" = "true" ]; then
-	pushd "$LIGHTNING_PATH/lightning-persister"
-	RUSTC_BOOTSTRAP=1 cargo rustc --profile=check -- -Zunstable-options --pretty=expanded > /tmp/lightning-persist-crate-source.txt
+function add_crate() {
+	pushd "$LIGHTNING_PATH/$1"
+	RUSTC_BOOTSTRAP=1 cargo rustc --profile=check $3 -- -Zunstable-options --pretty=expanded > /tmp/$1-crate-source.txt
 	popd
 	if [ "$HOST_PLATFORM" = "host: x86_64-apple-darwin" ]; then
-		sed -i".original" '1i\
-pub mod lightning_persister {
-' /tmp/lightning-persist-crate-source.txt
+		sed -i".original" "1i\
```

Comment:
Need an extra `\`:

```
sed -i".original" "1i\\
```

Reply:
Done and squashed into "Simplify crates in genbindings, add invoice, and bump upstream git" (which already touched this line).